### ISSUE TITLE
`DataLoader` doesn't respond to `set_default_tensor_type`

### ISSUE THREAD

**Issue by eulerreich** [pytorch/pytorch#10393]
If you have a question or would like help and support, please ask at our
[forums](https://discuss.pytorch.org/).

If you are submitting a feature request, please preface the title with [feature request].
If you are submitting a bug report, please fill in the following details.

## Issue description

If I set default tensor to double, `DataLoader` still yields float tensors.

## Code example

```python
import torch
import torch as th
import torch.utils.data as tdata
from torch import nn
from torchvision import datasets, transforms

th.set_default_dtype(torch.float64)
th.set_default_tensor_type(torch.DoubleTensor)

train_data = datasets.CIFAR10('../data', train=True, download=True, transform=transforms.ToTensor())
train_loader = tdata.DataLoader(train_data, batch_size=1000, shuffle=True)

next(iter(train_loader))[0].dtype
```
## System Info
Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```
PyTorch version: 0.4.1
Is debug build: No
CUDA used to build PyTorch: 9.0.176

OS: Ubuntu 16.04.4 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.10) 5.4.0 20160609
CMake version: version 3.5.1

Python version: 2.7
Is CUDA available: Yes
CUDA runtime version: Could not collect
GPU models and configuration: GPU 0: Tesla P100-PCIE-16GB
Nvidia driver version: 390.12
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.7.1.2

Versions of relevant libraries:
[pip] numpy (1.14.3)
[pip] numpydoc (0.8.0)
[pip] torch (0.4.1)
[pip] torchvision (0.2.1)
[conda] pytorch                   0.4.1           py27_cuda9.0.176_cudnn7.1.2_1    pytorch
[conda] torchvision               0.2.1                    py27_1    pytorch

**Comment by ArmenAg** [comment]
I don't think this is an issue. datasets.CIFAR10 loads up the data into a numpy array (https://github.com/pytorch/vision/blob/master/torchvision/datasets/cifar.py#L95). The default collate functionality for collating numpy arrays is simply to use torch.from_numpy() which preserves type (as it should).
```

if elem_type.__name__ == 'ndarray':
            # array of string classes and object
            if re.search('[SaUO]', elem.dtype.str) is not None:
                raise TypeError(error_msg.format(elem.dtype))

            return torch.stack([torch.from_numpy(b) for b in batch], 0)
```

**Comment by ngimel** [comment]
Don't know if this issue still exists.

**Comment by ejguan** [comment]
Can confirm the inconsistent behavior, but the problem comes from `torchvision.transform.ToTensor()`
https://github.com/pytorch/vision/blob/af97ec2f4c9daac091b9a87355c4f22d37488004/torchvision/transforms/functional.py#L133-L140
```py
import numpy as np
import torch
from PIL import Image
from torchvision import transforms


np_arr = np.random.randint(0, 255, (32,32,3), dtype=np.uint8)
img = Image.fromarray(np_arr)
trans = transforms.ToTensor()
res = trans(img)
print(res.dtype) # torch.float32
```
